tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_full.py

```diff
@@ -31,6 +32,16 @@ def example_files(tmp_path: Path) -> Path:
     return artifact_dir
 
 
+@pytest.fixture
+def logged_artifact(wandb_init, example_files) -> Artifact:
+    with wandb.init() as run:
+        artifact = wandb.Artifact("test-artifact", "dataset")
+        artifact.add_dir(example_files)
+        run.log_artifact(artifact)
+    artifact.wait()
```

Comment:
Just a note:
This is surprising. I would expect everything to have finished uploading when `wandb.finish()` (implicitly called at the end of the context above) returns.

Reply:
It's *very* surprising, and you're right, it shouldn't happen. I haven't dug into it, so I'm not sure if this is a limitation of our mocks (`wandb_init`) or is an actual bug in production, but yeah, the `wait()` call should be superfluous and isn't.